### PR TITLE
Implement `from_epoch(x:duration) -> time`

### PR DESCRIPTION
**Pull request by raxyte**
- Closes https://github.com/tenzir/issues/issues/2387

### CODE REVIEW

File: libtenzir/builtins/functions/time.cpp [hunk 1]
```diff
@@ -126,74 +126,42 @@ class since_epoch final : public function_plugin {
   }
 };
 
-class from_epoch_ms final : public function_plugin {
+class from_epoch final : public function_plugin {
```
Comment (jachris) [inline libtenzir/builtins/functions/time.cpp:129]: Maybe deprecate instead of directly remove?

File: libtenzir/builtins/functions/time.cpp [hunk 2]
```diff
@@ -126,74 +126,42 @@ class since_epoch final : public function_plugin {
...
-              append_builder(time_type{}, *b,
-                             time{duration{int64_t(
-                               value_at(double_type{}, arg, i) * 1'000'000)}}));
+            check(b->Append(arg.Value(i)));
           }
           return series{time_type{}, finish(*b)};
         },
         [&](const auto&) {
-          diagnostic::warning("`{}` expected number, but got `{}`", name(),
+          diagnostic::warning("`{}` expected `durationo`, but got `{}`", name(),
```
Comment (jachris) [inline libtenzir/builtins/functions/time.cpp:160]:
```suggestion
          diagnostic::warning("`{}` expected `duration`, but got `{}`", name(),
```

File: web/docs/tql2/functions.md [hunk 3]
```diff
@@ -126,7 +126,7 @@ Function | Description | Example
 Function | Description | Example
 :--------|:-------------|:-------
 [`as_secs`](functions/as_secs.md) | Converts a duration into seconds | `as_secs(42ms)`
-[`from_epoch_ms`](functions/from_epoch_ms.md) | Interprets a number as Unix time | `from_epoch_ms(1730234246123.456)`
+[`from_epoch`](functions/from_epoch.md) | Interprets a duration as Unix time | `from_epoch(10y)`
```
Comment (jachris) [inline web/docs/tql2/functions.md:129]: A better example would be a reasonable timestamp (or maybe just an appropriately named field).

File: web/docs/tql2/functions/from_epoch.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+Interprets a duration as Unix time.
+
+```tql
+from_epoch(x:duration) -> time
+```
+
+## Description
+
+The `from_epoch_ms` function interprets a number as [Unix
+time](https://en.wikipedia.org/wiki/Unix_time) in milliseconds.
```
Comment (jachris) [inline web/docs/tql2/functions/from_epoch.md:12]: Still says number here.

File: web/docs/tql2/functions/from_epoch.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+### `x: duration`
+
+The duration since the Unix epoch, i.e., 00:00:00 UTC on January
+1970.
```
Comment (jachris) [inline web/docs/tql2/functions/from_epoch.md:17]: That's missing that it's the first of January.

File: libtenzir/builtins/functions/time.cpp (hunk 1 above)
Comment (raxyte) [reply libtenzir/builtins/functions/time.cpp:129]: It's a simple upgrade path, still so?

Comment (dominiklohmann) [reply libtenzir/builtins/functions/time.cpp:129]: I suspect that this isn't used widely, so I'd be fine with just removing it.

File: web/docs/tql2/functions.md (hunk 3 above)
Comment (dominiklohmann) [reply web/docs/tql2/functions.md:129]: I think something like `from_epoch(time_ms * 1ms)` would be a reasonable example.

File: web/docs/tql2/functions/from_epoch.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+Interprets a duration as Unix time.
+
+```tql
+from_epoch(x:duration) -> time
+```
+
+## Description
+
+The `from_epoch` function interprets a duration as [Unix
+time](https://en.wikipedia.org/wiki/Unix_time) in milliseconds.
```
Comment (jachris) [inline web/docs/tql2/functions/from_epoch.md:12]:
```suggestion
The `from_epoch` function interprets a duration as [Unix
time](https://en.wikipedia.org/wiki/Unix_time).
```

**Review by jachris — APPROVED** [review]

**Pull request merged**